add build.zig.zon

Sivecano opened the pull request:
zig has a built-in package manager, and it's nice if we're able to comply with its requirements.
this file is mostly based on the `gyro.zzz` file already present. 
Tell me if this is somehow incomplete.

Sivecano commented:
I see tests failing. 
I'll need to test this with a more recent zig build. 
0.14 releases tomorrow. I'll see then.

ikskuh commented on build.zig.zon:
```diff
@@ -0,0 +1,11 @@
+.{
+    .name = "SDL",
```
`.name = .SDL2`

Sivecano replied:
applied this

Sivecano commented:
yep, definitely one for tomorrow

Sivecano commented:
I see, there's a field for a security fingerprint. 
I don't know how this is generated.
it may be better for @ikskuh (the actual manintainer) to generate it.

ikskuh commented:
> I see, there's a field for a security fingerprint. I don't know how this is generated. it may be better for @ikskuh (the actual manintainer) to generate it.

`zig build` will generate one for you. it's not a security fingerprint, just a random number to allow distinction of equally named packages

Sivecano commented:
now that 0.14.0 is finally out, I added the fingerprint and also fixed some functions in `build.zig` that weren't compiling before.
I've run all the examples with the 0.14.0 compiler.

Sivecano commented:
the macos builds are not passing. don't know why
